src/main/java/core/basesyntax/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,19 @@
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final Map<String, Integer> fruitStorage;
+
+    public ReportGeneratorImpl(Map<String, Integer> fruitStorage) {
```

Comment:
The constructor should check if the provided 'fruitStorage' map is not null to avoid potential NullPointerExceptions in the getReport method.